acestep/api_server.py

```
@@ -540,6 +540,297 @@ class Config:
+                str(tensorboard_port),
+                "--bind_all",
+            ],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+        )
+        return f"http://localhost:{tensorboard_port}"
+    except Exception:
+        return None
+
+
+@contextmanager
```

Ruff (0.15.0)

[error] Undefined name `contextmanager`

(F821)